mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h

```diff
@@ -45,7 +55,8 @@ class GreedyRewriteConfig {
   /// patterns.
   ///
   /// Note: Only applicable when simplifying entire regions.
-  bool enableRegionSimplification = true;
+  GreedySimplifyRegionLevel enableRegionSimplification =
+      GreedySimplifyRegionLevel::Aggressive;
```

Comment:
Sorry, I didn't spot this, but I think this should be set as `Normal`. Having `Aggressive` by default will make the compiler hang for large merges.

Reply:
This is preserving the behavior of the GreedyPatternRewriter.
This is not the default for the canonicalize pass.

Reply:
The point is that the initial bug was seen in the rewriter (canonicalization was the way I reproduced it). Maybe we could set a threshold for the number of block arguments to merge (like I was proposing above)?

Reply:
> The point is that the initial bug was seen in the rewriter 

My point is that clients of the rewriter for purposes other than canonicalization should decide what they want. I'm rather confident in changing this default for the purpose of canonicalization, I don't think I need to change other usage of the greedy driver right now.

>  Maybe we could set a threshold for the number of block arguments to merge (like I was proposing above)?

That seems like a clunk to me to be honest. I would rather investigate the issue first.

Reply:
Ok, that's fine. I will try to work on the core issue, and meanwhile we can configure the rewriter to do a `Normal` region simplification.

Reply:
Thanks @joker-eph